### PR TITLE
Work out audio upload issues

### PR DESCRIPTION
**Pull request by CharlieMcVicker**
- fix backend weirdness related to merge?
- cleanup contributor audio editing
- prettier

# TODO
- [x] Some AWS stuff is wonky: Uploads to S3 fail with `NotAuthorizedException: Token is not from a supported provider of this identity pool.`
- [x] The styles for this should be refactored
- [ ] It's possible `contributors.tsx` needs to be broken up

### CODE REVIEW

File: website/src/components/edit-word-audio/contributor.tsx [hunk 1]
```diff
@@ -121,35 +154,70 @@ export function ContributorEditWordAudio(p: {
...
+    }
+  }
+
+  return (
+    <>
       <VisuallyHidden>
         <input type="file" id="file-upload" onChange={onFileChanged} />
       </VisuallyHidden>
-      <div
+      <form
```
Comment (CharlieMcVicker) [inline website/src/components/edit-word-audio/contributor.tsx:200]: I'm not sure this change was actually needed

File: website/src/components/edit-word-audio/contributor.tsx [hunk 2]
```diff
@@ -121,35 +154,70 @@ export function ContributorEditWordAudio(p: {
...
+            as="label"
+            htmlFor="file-upload"
             className={
               currentTab === "upload" ? subtleButtonActive : subtleButton
             }
           >
             Upload audio
           </IconTextButton>
-        </label>
+        </div>
```
Comment (CharlieMcVicker) [inline website/src/components/edit-word-audio/contributor.tsx:220]: There was some bad interaction happening between the button and label elements here. This seems to have resolved the issue, kept the tab order correct, and kept cursor changes correct as well.

File: website/src/components/edit-word-audio/contributor.tsx [hunk 3]
```diff
@@ -163,18 +231,58 @@ export function ContributorEditWordAudio(p: {
...
+        left: 0,
+        bottom: 0,
+        right: 0,
+        background: "rgba(255,255,255,0.65)",
+        color: "black",
+        padding: 40,
+        backdropFilter: "blur(2px)",
+        border: `4px solid ${border}`,
+      }}
+    >
```
Comment (CharlieMcVicker) [inline website/src/components/edit-word-audio/contributor.tsx:284]: This should live in a `...styles.tsx` file

File: website/src/components/edit-word-audio/contributor.tsx [hunk 4]
```diff
@@ -325,5 +433,5 @@ async function uploadContributorAudioToS3(user: CognitoUser, data: Blob) {
     })
   )
 
-  return { resourceUrl: `https://d1q0qkah8ttfau.cloudfront.net/${key}` }
+  return { resourceUrl: `https://${process.env["CF_URL"]}/${key}` }
```
Comment (CharlieMcVicker) [inline website/src/components/edit-word-audio/contributor.tsx:436]: oops 😶

File: website/src/components/wordpress.tsx [hunk 5]
```diff
@@ -81,7 +81,8 @@ const parseOptions: HTMLReactParserOptions = {
   replace(node) {
     if ("data" in node) {
       const referenceEmbedStyle = /\[(\w*)\]/ // [search | glossary]
-      const wordEmbedStyle = /\[(\w*):([0-9]*)-?([0-9]*)?:?(audio)?(join)?(#)?(\w*)?\]/ // [DocName:Start(-OptionalEnd):?(audio?)(join?)#?OptionalChapterSlug?]
+      const wordEmbedStyle =
+        /\[(\w*):([0-9]*)-?([0-9]*)?:?(audio)?(join)?(#)?(\w*)?\]/ // [DocName:Start(-OptionalEnd):?(audio?)(join?)#?OptionalChapterSlug?]
```
Comment (CharlieMcVicker) [inline website/src/components/wordpress.tsx:85]: @GracefulLemming we may be in a formatting battle - my `dev-check` changed this, but the last edit was you and the commit message was `prettier`

**Review by CharlieMcVicker — COMMENTED** [review]
Roadmap of changes

File: website/src/components/wordpress.tsx (hunk 5 above)
Comment (GracefulLemming) [reply website/src/components/wordpress.tsx:85]: try merging main in– running `prettier -w src` and `prettier -w --config package.json src` yields two different outputs. All of our checks use the one with `--config` specified so thats likely what I ran

**Comment by GracefulLemming** [comment]
https://github.com/NEU-DSG/dailp-encoding/blob/863bf0dfd7b6e33a0cb9a28773e76f4ced3b61a9/terraform/user-roles.nix#L26
Ah... fixing this in a new branch now

File: website/src/components/edit-word-audio/contributor.tsx [hunk 6]
```diff
@@ -121,35 +154,70 @@ export function ContributorEditWordAudio(p: {
...
+    const [file] = event.currentTarget.files
+    if (!file) return
+    console.log("New file selected", file)
+    setSelectedFile(file)
+    setCurrentTab("upload")
+  }
+
+  async function uploadAudioAndReset(data: Blob) {
+    const success = await uploadAudio(data)
+    if (success) {
```
Comment (GracefulLemming) [inline website/src/components/edit-word-audio/contributor.tsx:189]: Should we notify the user if this fails, or at least log an error in the console?

File: website/src/components/edit-word-audio/contributor.tsx (hunk 3 above)
Comment (GracefulLemming) [reply website/src/components/edit-word-audio/contributor.tsx:284]: Please make this change

**Review by GracefulLemming — CHANGES_REQUESTED** [review]

File: website/src/components/edit-word-audio/contributor.tsx [hunk 7]
```diff
@@ -323,5 +418,5 @@ async function uploadContributorAudioToS3(user: CognitoUser, data: Blob) {
     })
   )
 
-  return { resourceUrl: `https://d1q0qkah8ttfau.cloudfront.net/${key}` }
+  return { resourceUrl: `https://${process.env["CF_URL"]}/${key}` }
```
Comment (CharlieMcVicker) [inline website/src/components/edit-word-audio/contributor.tsx:421]: Does the cloudflare url have `https://` at the front or no? I found some code from @GracefulLemming from four months ago where we started prepending `https://` in a backend process. I'm struggling to find where this secret is managed.

**Review by GracefulLemming — APPROVED** [review]

**Pull request merged**